src/main/java/org/elasticsearch/index/translog/TranslogSnapshot.java

```diff
@@ -42,21 +42,19 @@
      * Create a snapshot of translog file channel. The length parameter should be consistent with totalOperations and point
      * at the end of the last operation in this snapshot.
      */
-    public TranslogSnapshot(List<ChannelSnapshot> orderedTranslogs, ESLogger logger) {
```

Comment:
Can we rename this class?  It's confusing to have Translog.Snapshot and TranslogSnapshot ... maybe MultiTranslogSnapshots (like Lucene's MultiReader), since it logically appends multiple snapshots?